### ISSUE TITLE
Weirdness with Go 1.13.3 and Go Modules

### ISSUE THREAD

**Issue by zoidbergwill** [bookingcom/shipper#230]
# Overview

*TL;DR*: To use Shipper as a dependency in my project I needed to add the following to my `go.mod`:

```
replace k8s.io/apimachinery v0.0.0-20190602113612-63a6072eb563 => k8s.io/apimachinery v0.0.0-20190602183612-63a6072eb563
```

# Expected Behaviour

If I `go get -u github.com/bookingcom/shipper` or manually clone the repo and run `go build cmd/shipper/main.go`, I expect it to work.

# Actual Behaviour

After fetching the repo manually, and trying to run a `go build` or `go mod tidy`, I get an error about the `k8s.io/api` commit we require, because of the `k8s.io/apimachinery` commit that the mentioned `k8s.io/api` commit requires.

```
# go build cmd/shipper/main.go
go: finding k8s.io/apimachinery 63a6072eb563
go: k8s.io/api@v0.0.0-20190602205700-9b8cae951d65 requires
        k8s.io/apimachinery@v0.0.0-20190602113612-63a6072eb563: invalid pseudo-version: does not match version-control timestamp (2019-06-02T18:36:12Z)
```

# Potential Solutions

## Updating go.mod to use the replace

# Notes

@icanhazbroccoli and I investigated it a bit, and it seems to be pretty common with folks with some versions of Go 1.13 and using go modules.

[Go 1.13 Version Validation](https://golang.org/doc/go1.13#version-validation)
